transmonee_dashboard/src/transmonee_dashboard/pages/base_page.py

```diff
@@ -899,11 +899,18 @@ def main_figure(indicator, selections, indicators_dict):
         )  # Add sorting by Year to display the years in proper order
         .reset_index()
     )
-
-    # print("Sorted Data", df)
     options["labels"] = DEFAULT_LABELS.copy()
     options["labels"]["OBS_VALUE"] = name
-    return px.scatter_mapbox(df, **options), source
+    main_figure = px.scatter_mapbox(df, **options)
+    # set last frame as the active one; i.e. select the max year as the default displayed year
+    main_figure.layout["sliders"][0]["active"] = len(main_figure.frames) - 1
```

Comment:
This is good, but just do an if to check to make sure sliders are indeed in the layout, because we may not always define the option in the config and this could throw a key error here, something like this:

    if 'sliders' in main_figure.layout:
        main_figure.layout["sliders"][0]["active"] = len(main_figure.frames) - 1
        main_figure = go.Figure(
            data=main_figure["frames"][-1]["data"],
            frames=main_figure["frames"],
            layout=main_figure.layout,
        )

Reply:
Thanks!

I added this condition: if len(main_figure.layout["sliders"]) > 0; because the above didn't stop the exception as it returned true even when I remove the animation_frame from the config.

One thing to pay attention to is in case this animation_frame is not there, we need to aggregate the code to be properly displayed on the maps :).